curator-client/src/main/java/org/apache/curator/CuratorZookeeperClient.java

```diff
@@ -399,6 +399,9 @@ void removeParentWatcher(Watcher watcher) {
     public void internalBlockUntilConnectedOrTimedOut() throws InterruptedException {
         long waitTimeMs = connectionTimeoutMs;
         while (!state.isConnected() && (waitTimeMs > 0)) {
+            if (!started.get()) {
+                throw new IllegalStateException("Client is not started");
```

Comment:
+1 to this

This should cover both "not started" and "closed".

Reply:
Sounds good to me.